### PR TITLE
feat(table): add box-shadow on vertical scroll

### PR DESCRIPTION
**Pull request by Lenderson**
#### What is the purpose of this pull request?
Add box-shadow on vertical and horizontal scroll inside the table

#### How should this be manually tested?
Check the horizontal and vertical scroll stories

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/table/hooks/use-vertical-scroll.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+import { useStateContext } from '../context'
+
+export function useVerticalScroll() {
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/hooks/use-vertical-scroll.ts:4]: same for the file name

```suggestion
export function useTableScroll() {
```

File: packages/admin-ui/src/table/components/table-head.tsx [hunk 2]
```diff
@@ -17,8 +22,11 @@ export const TableHead = createComponent<'div'>((props) => {
   } as any
 
   return useElement('div', {
-    baseStyle: styles.baseline,
     ...headProps,
+    baseStyle: styles.baseline,
+    csx: {
+      boxShadow: hasVerticalScroll ? '$overlay.center' : 'none',
+    },
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/components/table-head.tsx:29]: you can add this as a variant to the `baseStyle`

```ts
baseStyle: {
  ...styles.baseline,
  ...style.variant({ hasVerticalScroll })
}
```

File: packages/admin-ui/src/table/components/table-cell.tsx [hunk 3]
```diff
@@ -29,12 +30,30 @@ export const TableCell = createComponent<'div', CellOptions>((props) => {
...
+      return
     }
+
+    const isBodyCell = ref.current.getAttribute('role') === 'cell'
+
+    const table = tableRef.current
+
+    ref.current.style.left = isBodyCell
+      ? `${ref.current.offsetLeft - table.offsetLeft}px`
+      : `${ref.current.offsetLeft}px`
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/components/table-cell.tsx:56]: I'm not sure if i get what this does exactly

**Review by lucasaarcoverde — COMMENTED** [review]
For some reason the table header is not sticky to me in the Vertical only example, is that right?

File: packages/admin-ui/src/table/components/table-head.tsx (hunk 2 above)
Comment (lucasaarcoverde) [reply packages/admin-ui/src/table/components/table-head.tsx:29]: Same in the horizontal usage

File: packages/admin-ui/src/table/components/table-cell.tsx (hunk 3 above)
Comment (Lenderson) [reply packages/admin-ui/src/table/components/table-cell.tsx:56]: I will put a comment to explain it  ;)

**Review by lucasaarcoverde — APPROVED** [review]
LGTM

**Pull request merged**